### PR TITLE
feat(lsp): more annotations

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

**Review by mfussenegger — APPROVED** [review]
👍

File: runtime/lua/vim/lsp/semantic_tokens.lua [hunk 1]
```diff
@@ -89,20 +91,24 @@ end
 
 --- Converts a raw token list to a list of highlight ranges used by the on_win callback
 ---
+---@param data any[]
```
Comment (jdrouhard) [inline runtime/lua/vim/lsp/semantic_tokens.lua:94]:
```suggestion
---@param data integer[]
```

File: runtime/lua/vim/lsp/semantic_tokens.lua [hunk 2]
```diff
@@ -89,20 +91,24 @@ end
...
+---@param bufnr integer
+---@param client lsp.Client
+---@param request any
```
Comment (jdrouhard) [inline runtime/lua/vim/lsp/semantic_tokens.lua:97]:
```suggestion
---@param request STActiveRequest
```

**Review by jdrouhard — COMMENTED** [review]
Awesome, thanks for cleaning up the annotations in the semantic tokens module!

File: runtime/lua/vim/lsp/semantic_tokens.lua (hunk 1 above)
Comment (lewis6991) [reply runtime/lua/vim/lsp/semantic_tokens.lua:94]: This causes a warning on line 132:

```lua
    local delta_line = data[i]
    line = line and line + delta_line or delta_line
```

Comment (lewis6991) [reply runtime/lua/vim/lsp/semantic_tokens.lua:94]: Ah fixed, `line` is an integer, not a string.

**Pull request merged**